core/app/components/DataTable/PubsDataTable/columns.tsx

```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import type { ColumnDef } from "@tanstack/react-table";
+
+import * as React from "react";
+
+import type { ProcessedPub } from "contracts";
+import type { DataTableRowAction } from "ui/data-table-paged";
+import { Checkbox } from "ui/checkbox";
+import { DataTableColumnHeader } from "ui/data-table";
+
+import { dateFormatOptions } from "~/lib/dates";
+import { getPubTitle } from "~/lib/pubs";
+
+interface GetColumnsProps {
+	setRowAction: React.Dispatch<
+		React.SetStateAction<DataTableRowAction<ProcessedPub<{ withPubType: true }>> | null>
+	>;
+}
+
+export function getColumns({
+	setRowAction,
+}: GetColumnsProps): ColumnDef<ProcessedPub<{ withPubType: true }>>[] {
+	return [
+		{
+			id: "select",
+			header: ({ table }) => (
+				<Checkbox
+					checked={
+						table.getIsAllPageRowsSelected() ||
+						(table.getIsSomePageRowsSelected() && "indeterminate")
+					}
+					onCheckedChange={(value) => table.toggleAllPageRowsSelected(!!value)}
+					aria-label="Select all"
+					className="translate-y-0.5"
+				/>
+			),
+			cell: ({ row }) => (
+				<Checkbox
+					checked={row.getIsSelected()}
+					onCheckedChange={(value) => row.toggleSelected(!!value)}
+					aria-label="Select row"
+					className="translate-y-0.5"
+				/>
+			),
+			enableSorting: false,
+			enableHiding: false,
+		},
```

Comment:
idk why but in all the screenshots you shared the checkbox column seems so wide, is that bc of the translate?